### PR TITLE
[DOC] upgrade mxnet dependency to 1.5.0

### PR DESCRIPTION
**Pull request by szha**
## Description ##
upgrade mxnet dependency to 1.5.0

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] upgrade mxnet dependency to 1.5.0

### CODE REVIEW

**Comment by mli** [comment]
Found link check problems in job PR-842/3:
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-842/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-842/3/index.html

**Comment by leezu** [comment]
@eric-haibin-lin there is an issue with float16 support for the bert finetune scripst on mxnet 1.5 `mxnet.base.MXNetError: [15:30:30] src/operator/tensor/./ordering_op-inl.h:551: This operation does not support float16`

**Comment by eric-haibin-lin** [comment]
There was a fix for AMP which is added to mxnet after 1.5 code freeze. I can help fix the script.

**Comment by mli** [comment]
Found link check problems in job PR-842/4:
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-842/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-842/4/index.html

**Comment by mli** [comment]
Found link check problems in job PR-842/5:
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-842/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-842/5/index.html

**Comment by leezu** [comment]
@szha could you please update cuda on the CI server. The current version is untested for mxnet 1.5 and may be responsible for the out of memory errors
http://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-gpu-integration/detail/PR-848/7/

File: env/gpu/py3-master.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ dependencies:
   - regex
   - pip:
     - pylint-quotes<0.2
-    - mxnet-cu92mkl>=1.5.0b20190407
+    - mxnet-cu101mkl>=1.5.0
```
Comment (szha) [inline env/gpu/py3-master.yml]: this should be a nightly version.

Comment (leezu) [reply env/gpu/py3-master.yml]: Ie the `master` in `py3-master.yml` refers to MXNet master branch?

**Comment by mli** [comment]
Found link check problems in job PR-842/13:
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-842/13 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-842/13/index.html

**Comment by mli** [comment]
Found link check problems in job PR-842/15:
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-842/15 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-842/15/index.html

File: env/gpu/py3-master.yml (hunk 1 above)
Comment (szha) [reply env/gpu/py3-master.yml]: yes

**Comment by mli** [comment]
Found link check problems in job PR-842/16:
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-842/16 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-842/16/index.html

**Pull request merged**